single-program/near-interface.js

```diff
@@ -0,0 +1,32 @@
+import { providers } from "near-api-js";
+import give_rewards from "xtoearn-near-sdk";
+
+function getConnInfo() {
+  const networkId = process.env.NEAR_ENV;
+  return { url: `https://rpc.${networkId}.near.org` };
+}
+
+async function getTransactionDetails(transactionHashes) {
+  const connInfo = getConnInfo();
+  const provider = new providers.JsonRpcProvider(connInfo);
+  const transaction = await provider.txStatus(transactionHashes, "unnused");
+  return transaction;
+}
+
+export default async function giveReward(userAccountId) {
+  const args = {
+    reward_name: "video_likes",
+    program_name: "pe",
+    user_wallet: userAccountId,
+    program_owner: process.env.ACCOUNT_ID,
+    private_key: process.env.PRIVATE_KEY,
+  };
+  const reward = await give_rewards(

```

Comment:
Can we display status message without querying the transaction (  getTransactionDetails ).

Reply:
Will check with Vaulstein